### ISSUE TITLE
[documentation]: live editable samples

### ISSUE THREAD

**Issue by olaf-k** [Vonage/vivid-3#1019]
## Goal
Replace the current static copiable code samples
![image](https://user-images.githubusercontent.com/1153034/217474689-852a3c7d-7e98-470e-a41d-58a47ce1d9c5.png)
by a minimalist inline editor to allow doc users to experiment without having to navigate away.
Experimenting with code samples on a dedicated platform like CodePen is already feasible and has been implemented in #1005 

## Process
1. [x] Select an editor.
2. [x] Integrate it in the doc build process.
3. [x] Replace the static samples.

### Selecting an editor
There are several solutions available at the time of writing. The following were considered:
- [ckeditor](https://ckeditor.com/): aimed at documents editing rather than code, commercial.
- [Monaco](https://microsoft.github.io/monaco-editor/): the editor used in VSCode.
- [Ace](https://ace.c9.io/): one of the oldest editor published.
- [CodeMirror](https://codemirror.net/): another old editor, refactored in its current v6.

Due to its nature as a document editor rather than code-oriented, ckeditor has been discarded early.
While interesting technically, Monaco unfortunately falls short on mobile support and as such has also been discarded.
![image](https://user-images.githubusercontent.com/1153034/217477958-2aad66b4-6584-4268-a625-95e8c9d26a48.png)
There is [an interesting article written by the people at replit](https://blog.replit.com/code-editors) about their choice of an online editor. 
Long story short: let's try CodeMirror.

### Integration in the doc app
Today, the doc app is not an individual app per se: it doesn't have a `package.json` and publishes itself as a static web site. The JS code used in the app is either:
- used at build time
- published as assets to be run in the browser

These assets are simple and are copied as-is, there is no bundling process in place. This is needed for CodeMirror (and possibly other editors installed via npm). A quick workaround using rollup has been done for the PoC and will need to be replaced for prod (proper nx integration and minification as the current bundle weighs ~1MB!).

### Replace the static samples
![chrome_k7vBZkZbDH](https://user-images.githubusercontent.com/1153034/217609755-6528e573-2c8c-43bf-af33-1ccc5b4931a3.gif)

**Comment by yinonov** [comment]
how needed are editors within mobile platforms? could be nice to view but it's hardly used with copy & paste, or even less likely edited(?)
Mobile can fallback to simply showing the code

**Comment by olaf-k** [comment]
> how needed are editors within mobile platforms? could be nice to view but it's hardly used with copy & paste, or even less likely edited(?) Mobile can fallback to simply showing the code

I personally wouldn't but know people who don't mind coding on an iPad 🤷‍♀️
In out particular case, it wouldn't hurt having a solution that works on all devices considering live samples are very limited in code size and functional scope. Not mentioning the fact that adding a fencing mechanism to target only non-mobile users makes its more complicated.

Note that the main example on [Monaco's homepage](https://microsoft.github.io/monaco-editor/) seems to behave fine on an iPhone. I'm guessing "no mobile support" just means "if it breaks there we won't fix it".

**Comment by olaf-k** [comment]
Update:

I wanted to move the code examples display at runtime for reasons:
- remove the build code that creates files for them
- remove duplication (the code samples are stored in the `<pre>` and in said files)
- prepare for the actual code editor PR (since edited code will need to be updated live)

[You can browse the resulting code here](https://github.com/Vonage/vivid-3/commit/fdf79d233ebe8245bc67469d70fe30d30c69970f), in the `load-samples-runtime` branch.

Unfortunately, I won't propose a PR as this comes with display bugs I discovered late in the process.
- The tooltip suffers from a FoUT (it's subtle but it's here):
![tooltipFOUT](https://user-images.githubusercontent.com/1153034/218266079-b8839e9c-c581-4db5-a60b-b29e15ba48ac.gif)
- The popup arrow sample is not centered:
![popupuncentered](https://user-images.githubusercontent.com/1153034/218266108-72c81894-97a5-47ab-813c-530417c12403.png)
- The code also seems to drive the `autoresize()` function crazy in the color-palette page sample:
![weirdresizebug](https://user-images.githubusercontent.com/1153034/218266175-156995e1-470e-4f5f-8c63-9c4265b9b18d.gif)

This is disappointing and I'll need to make sure the code updates made from the editor are displayed correctly before pushing its implementation.
